### DISCUSSION TITLE
getting my head around classifications

### DISCUSSION THREAD

**Discussion by mophilly** [enoch3712/ExtractThinker#147] [General]
I have a basic document flow working... on just one example document. :-)
The first set of documents contain the same data elements as the initial example but have different layouts. I have identified five distinct layouts and I suspect the variants will number over a dozen in short order.

I would like to learn more about classification and strategy to achieve a reliable result. A web search on the topic returns dozens of links. I am reading a few, but I would like my time spent in a well focused manner.

I would love some links to articles and similar that are aligned with this project.

**Comment by enoch3712** [comment]
Hello @mophilly! Happy holidays! 

What you want basically several types of similar types of documents correct? 

You can use ExtractThinker Classification, is perfect for this:

Example of classification:
```
    # Create classifications
    classifications = [
        Classification(
            name="Vehicle Registration",
            description="This is a vehicle registration document",
            contract=VehicleRegistration,
            extractor=extractor
        ),
        Classification(
            name="Driver License",
            description="This is a driver license document",
            contract=DriverLicense,
            extractor=extractor
        )
    ]

result = process.classify(DRIVER_LICENSE_FILE_PATH, COMMON_CLASSIFICATIONS, strategy=ClassificationStrategy.CONSENSUS)
```

The Type will be injected inside of the prompt, so will pick the best classification and you can just extract right away. 

Image that is a type of document like a form, with a predicted look and not only fields, you can inject an example of an image:

```
    # Create classifications
    classifications = [
        Classification(
            name="Vehicle Registration",
            description="This is a vehicle registration document",
            contract=VehicleRegistration,
            extractor=extractor,
            image=imageForm1
        ),
        Classification(
            name="Driver License",
            description="This is a driver license document",
            contract=DriverLicense,
            extractor=extractor,
            image=imageForm2
        )
    ]

result = process.classify(DRIVER_LICENSE_FILE_PATH, COMMON_CLASSIFICATIONS, strategy=ClassificationStrategy.CONSENSUS, image=True)
```

I advice you to look around, because according to clients and use cases, it fits 99% of them. You can see the group of tests here for better classification: 

https://github.com/enoch3712/ExtractThinker/blob/main/tests/test_classify.py


Types of articles, about classification with LLMs, this is the root of the work: 

https://python.useinstructor.com/examples/classification/

Then you have my articles that is an evolution of this for Document Intelligence :

https://medium.com/gitconnected/advanced-document-classification-with-llms-8801eaee3c58
https://medium.com/gitconnected/classification-tree-for-llms-32b69015c5e0


Dont forget you have the docs about this:
https://enoch3712.github.io/ExtractThinker/core-concepts/classification/


Hope it helps :)
